Added some of the stuff you suggested

a-Beanz opened the pull request:
Went ahead and took the time to add the stuff you suggested earlier.

a-Beanz committed:
Fix

Hunter fixed some stuff with the code made it overall better.

a-Beanz committed:
Added Suggested Items

Added stuff that was suggested like making a config to save the setting, located in the user's appdata local folder and making it uniform across some forms.

a-Beanz committed:
Merge branch 'Dark-Mode' of https://github.com/why1116/e-ReaderCardCreator into Dark-Mode

a-Beanz committed:
Merge branch 'master' into Dark-Mode

HunterRDev commented on AC_e-Reader_Card_Creator/CCC-Frontend.cs:
```diff
@@ -593,6 +632,332 @@ private void ToggleDarkMode(object sender, EventArgs e)
+            return false;
+        }
+
+        private void menu_View_Click(object sender, EventArgs e)
```
Remove extraneous empty function

HunterRDev commented on AC_e-Reader_Card_Creator/CCC-Frontend.cs:
```diff
@@ -593,6 +632,332 @@ private void ToggleDarkMode(object sender, EventArgs e)
+        }
+    }
+}
+ntains(letter_sender))
```
Fails to build due to all of this copy/pasted repeated code. Assuming this is a mistake

a-Beanz replied:
This wasn't on my local copy, interesting.

HunterRDev commented on AC_e-Reader_Card_Creator/CCC-Frontend.cs:
```diff
@@ -593,6 +632,332 @@ private void ToggleDarkMode(object sender, EventArgs e)
+            SaveDarkModePreference(isDarkModeEnabled);
+        }
+
+        private void ApplyDarkMode()
```
It'd be a good idea to outsource the contents of this function (and `ToggleDarkMode()` ) to a helper function in `References/Common.cs`. This way we don't have to repeat code for each form added.

HunterRDev commented on AC_e-Reader_Card_Creator/Print-Frontend.cs:
```diff
@@ -2,19 +2,132 @@
+            File.WriteAllText(preferenceFilePath, darkModeEnabled.ToString());
+        }
+
+        public class PrintFrontend
```
A new class shouldn't be necessary

HunterRDev commented on AC_e-Reader_Card_Creator/Print-Frontend.cs:
```diff
@@ -2,19 +2,132 @@
+            ApplyTheme();
+        }
+
+        private void ApplyTheme()
```
Why call this function if all it does is refer to another function later down? 🤔

HunterRDev commented on AC_e-Reader_Card_Creator/Print-Frontend.cs:
```diff
@@ -2,19 +2,132 @@
+            PrintFrontend.ApplyTheme(this, isDarkModeEnabled);
+        }
+
+        private void ToggleDarkMode(object sender, EventArgs e)
```
Repeating a previous comment, but all of this should probably be outsourced to a helper function so we don't have so much repeated copy/pasted code between each `Frontend.cs` form.

HunterRDev commented on AC_e-Reader_Card_Creator/CCC-Frontend.cs:
```diff
@@ -593,6 +632,332 @@ private void ToggleDarkMode(object sender, EventArgs e)
+
+        private void SaveDarkModePreference(bool darkModeEnabled)
+        {
+            string appDataFolderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "AC e-Reader Card Creator");
```
I suggest renaming the created directory and file as `e-ReaderCardCreator/config.txt`. Inside `config.txt`, we could have a `darkModePreference:` variable which can be read by the app. This way, we can add other variables to the config file if we add other user-specific settings in the future.

HunterRDev reviewed (changes requested):
I appreciate the work! However, there are some changes that need to be made here before merging:

- This code currently cannot build due to the inclusion of a couple hundred copy/pasted lines near the end of `CCC-Frontend.cs`
- I've added comments at various lines within this PR that I think would help. Most importantly, we should probably move handling the dark mode toggle to a helper function in `Common.cs` to reduce repetitive code across forms
- We should rename the created `.txt` file as just `config.txt` and handle toggling through a `darkModePreference` variable within said file. This gives us scalability for more saved config options in the future.
- The size of the main form seems broken; ideally fixed by handling the size dynamically for different resolutions and viewports, but that's beyond the scope of this PR.

a-Beanz committed:
Added Suggested Items

i'm tired

a-Beanz committed:
whoops

forgot to do something

why1116 commented:
All requested changes **SHOULD** be resolved, i may have missed one as I'm tired and haven't slept yet.

a-Beanz reviewed (commented):
All requested changes SHOULD be resolved, i may have missed one as I'm tired and haven't slept yet.

a-Beanz committed:
fixed window size

a-Beanz committed:
broke something

somehow i broke something

a-Beanz committed:
Update CCC-Frontend.Designer.cs

a-Beanz committed:
Update Print-Frontend.cs

a-Beanz committed:
Update CCC-Frontend.cs

a-Beanz committed:
Update CCC-Frontend.cs

a-Beanz committed:
Update Print-Frontend.cs

Hunter-Raff commented:
I'll review this over the weekend when I get a chance 👍

why1116 commented:
Alright, I'll take this time to look and patch any issues i come across/

a-Beanz committed:
Attempting a fix

Attempting a fix for a bug.

a-Beanz committed:
bwomp

a-Beanz committed:
Fixed a Bug

I found a bug where a white blob would cover the letter image when switching from dark mode to light mode, the issue has been fixed.

a-Beanz committed:
made another mistake

Accidentally deleted some code at some point and forgot, my bad

a-Beanz committed:
fixed ANOTHER issue

i'm sooo tired.

a-Beanz committed:
Update preferencesManager.cs

a-Beanz committed:
window size was poo poo

a-Beanz committed:
Update CCC-Frontend.cs

a-Beanz committed:
updated scripts
